hub/src/main/java/com/hub/HubController.java

```diff
@@ -124,14 +124,22 @@ protected void randomlyCheck() throws Exception {
                     .timeout(Duration.ofMillis(300))
                     .GET()
                     .build();
-            HttpResponse<String> response = HttpClient.newBuilder()
-                    .build()
-                    .send(request, HttpResponse.BodyHandlers.ofString());
+            HttpResponse<String> response = null;
+            try{
+                response = HttpClient.newBuilder()
+                        .build()
+                        .send(request, HttpResponse.BodyHandlers.ofString());
+            }catch(Exception e){
+                findLeader();
```

Comment:
in this case, we dont want to send the leader? I guess my question is - do we not always want to send leader when finding?